### PR TITLE
More FileHandler functionality

### PR DESCRIPTION
**Pull request by gonzalonunez**
- Implemented empty methods in `FileHandler`
- Wrote tests for bitfield comparisons
- Migrated `FileHandler` to an in-memory `HashMap<Integer, byte[]>`. Aggregating all pieces still saves it to a file.
- Wrote tests that cover 90% of methods in `FileHandler`

### CODE REVIEW

File: src/files/FileHandler.java [hunk 1]
```diff
@@ -120,31 +82,82 @@ public void chunkFile(){
...
             e.printStackTrace();
         }
     }
 
-    // TODO: Create function that returns the byte data for a piece given the index
-    public byte[] getPieceByteData(int pieceIndex){
-        return null;
+    // Returns the byte data for a piece given the index
+    public byte[] getPieceByteData(int pieceIndex) {
+        //FIXME: do we need to retrieve from an in-memory data store and if we fail only then fall back to on disk?
```
Comment (gonzalonunez) [inline src/files/FileHandler.java]: @nt92 ??

File: src/files/FileHandler.java [hunk 2]
```diff
@@ -120,31 +82,82 @@ public void chunkFile(){
...
+            e.printStackTrace();
+            return null;
+        }
     }
 
-    // TODO: Create function that adds a piece to the file's data given pieceData and the index
-    public void addPiece(int pieceIndex, byte[] pieceData){
+    // Adds a piece to the file's data given pieceData and the index
+    public void addPiece(int pieceIndex, byte[] pieceData) {
+        //FIXME: do we need to keep an in-memory data store of this as well?
```
Comment (gonzalonunez) [inline src/files/FileHandler.java]: @nt92 ??

File: src/files/FileHandler.java [hunk 3]
```diff
@@ -8,89 +8,53 @@
...
-        if(file.exists()){
-            return true;
-        }
-        return false;
+        return file.exists();
     }
 
-    public boolean hasAllPieces(){
+    public boolean hasAllPieces() {
+        //FIXME: do we need to retrieve from an in-memory data store and if we fail only then fall back to on disk?
```
Comment (gonzalonunez) [inline src/files/FileHandler.java]: @nt92 ??

File: src/files/FileHandler.java [hunk 4]
```diff
@@ -8,89 +8,53 @@
...
-                }
-            } catch(Exception e) {
-                e.printStackTrace();
-            }
-        }
-    }
-
-    public boolean hasPiece(int pieceIndex){
+    public boolean hasPiece(int pieceIndex) {
+        //FIXME: do we need to retrieve from an in-memory data store and if we fail only then fall back to on disk?
```
Comment (gonzalonunez) [inline src/files/FileHandler.java]: @nt92 ??

File: src/files/FileHandler.java (hunk 3 above)
Comment (takashiw) [reply src/files/FileHandler.java]: If that was is easier to implement, then we can. The original way I had it made saving to File the easier to do it

Comment (gonzalonunez) [reply src/files/FileHandler.java]: @takashiw Ok I'll go ahead and refactor this. We might do way too much I/O if we try to read/write to files all the time.

**Comment by gonzalonunez** [comment]
@nt92 @takashiw I refactored `FileHandler` over to an in-memory data store. Take a look at it whenever you get the chance? Just to make sure we're on the same page 🙏🏻

**Comment by gonzalonunez** [comment]
Merging. Ly fam

**Pull request merged**